packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1398,12 +1398,20 @@ class _SelectableFragment with Selectable, Diagnosticable, ChangeNotifier implem
       startSelectionPoint: SelectionPoint(
         localPosition: startOffsetInParagraphCoordinates,
         lineHeight: paragraph._textPainter.preferredLineHeight,
-        handleType: flipHandles ? TextSelectionHandleType.right : TextSelectionHandleType.left
+        handleType: selectionStart == selectionEnd
+                    ? TextSelectionHandleType.collapsed
+                    : flipHandles
+                    ? TextSelectionHandleType.right
+                    : TextSelectionHandleType.left,
```

Comment:
Nit: Would it be easier to read if you indented these two lines? Same for below.

Or you could always do a switch, but I'm not sure if that would help with readability or not here.

Reply:
I think I like the switch better [`48f01fc` (#157815)](https://github.com/flutter/flutter/pull/157815/commits/48f01fc8b07bb7eae7dc2775691bd126d953f816) .